k8s/k8s.go

```diff
@@ -396,6 +400,35 @@ func (k *k8sOps) RemoveLabelOnNode(name, key string) error {
 // which is invoked when the v1.Node object is changed.
 type NodeWatchFunc func(node *v1.Node) error
 
+// handleWatch is internal function that handles the Node-watch.  On channel shutdown (ie. stop watch),
+// it'll attempt to re-establish the its watch function.
+func (k *k8sOps) handleWatch(watchInterface watch.Interface, node *v1.Node, watchNodeFn NodeWatchFunc) {
+	for {
+		select {
+		case event, more := <-watchInterface.ResultChan():
+			if !more {
+				logrus.Warn("Kubernetes NodeWatch closed (attempting to reestablish)")
```

Comment:
nit: Can we only log when we know for sure that we failed to re-establish watch?
The reason being, I have seen some k8s clusters which had continuous watch failures hogging the system logs with these watch related logs.